src/Combinatorics/Matroids/ChowRings.jl

```diff
@@ -0,0 +1,217 @@
+        F = proper_flats[i]
+        poly = ring(0)
+        for index in findall(issubset(F,G) for G in proper_flats)
+            poly+= vars[index]
+        end
+        push!(relations, poly+vars[s+i]-vars[2*s+1]) #add h_F and subtract h_E
+    end
+    return relations
+end
+
+@doc Markdown.doc"""
+    augmented_chow_ring(M::Matroid)
+
+Return an augmented Chow ring of a matroid. As described in [BHMPW20](@cite). 
+
+# Example
+```jldoctest
+julia> M = fano_matroid();
+
+julia> R = augmented_chow_ring(M);
+```
+"""
+#This function was implemented by Fedor Glazov
+function augmented_chow_ring(M::Matroid)
+    Flats = flats(M)
+    sizeFlats = length(Flats)
+    n = size_groundset(M)
+
+    is_loopless(M) || error("Matroid has loops")
+    sizeFlats>3 || error("Matroid has too few flats")
+
+    proper_flats = Flats[1:sizeFlats-1]
+    element_var_names = [string("y_", S) for S in M.groundset]
+
+    flat_var_names = ["x_{" * join(S, ",") * "}" for S in proper_flats]
+    var_names = vcat(element_var_names, flat_var_names)
+    s = length(var_names)
+
+    ring, vars = PolynomialRing(QQ, var_names)
+    element_vars = vars[1:n]
+    flat_vars = vars[n+1:s]
+
+    I = augmented_linear_relations(ring, proper_flats, element_vars, flat_vars, M)
+    J = augmented_quadratic_relations(ring, proper_flats, element_vars, flat_vars, M)
+
+    chow_modulus = ideal(ring, vcat(I, J))
+    chow_ring, projection = quo(ring, chow_modulus)
+
+    return chow_ring
+end
+
+function augmented_linear_relations(ring::MPolyRing, proper_flats::Vector{Vector}, element_vars::Vector, flat_vars::Vector, M::Matroid)
+    n = size_groundset(M)
+    relations = Vector{elem_type(ring)}(undef,n)
+    i = 1
+    for element in M.groundset
+        relations[i] = element_vars[i]
+        j = 1
+        for proper_flat in proper_flats
+            if !(element in proper_flat)
+                relations[i] -= flat_vars[j]
+            end
+            j += 1
+        end
+        i += 1
+    end
+    return relations
+end
+
+function augmented_quadratic_relations(ring::MPolyRing, proper_flats::Vector{Vector}, element_vars::Vector, flat_vars::Vector, M::Matroid)
+    incomparable_polynomials = quadratic_relations(ring, proper_flats, flat_vars)
+    xy_polynomials = elem_type(ring)[]
+
+    i = 1
+    for element in M.groundset
+        j = 1
+        for proper_flat in proper_flats
+            if !(element in proper_flat)
+                push!(xy_polynomials, element_vars[i] * flat_vars[j])
+            end
+            j += 1
+        end
+        i += 1
+    end
+
+    return vcat(incomparable_polynomials, xy_polynomials)
+end
+
+"""
+A helper function to select indicies of a vector that do `include` elments of a given set and `exclude` anothers
+"""
+function _select(include::Union{AbstractVector,Set},exclude::Union{AbstractVector,Set},set::Union{AbstractVector,Set})
+    all = union(all...)
+    return findall(s->issubset(include,s)&&issubset(s,setdiff(all,exclude)),set);
```

Comment:
This recomputes `setdiff(all,exclude)` once for each element of `set` for which the first condition holds. So perhaps pull it out. Or perhaps avoid the potentially slow subset test, and just use something like `!any(occursin(exclude),s)`

Reply:
I pulled it out of the `findall()`